WordPress/Classes/ViewRelated/Post/PostEditor+MoreOptions.swift

```diff
@@ -23,33 +23,68 @@ extension PostEditor where Self: UIViewController {
         }
     }
 
-    private func savePostBeforePreview(completion: @escaping (() -> Void)) {
-        if post.isDraft() && post.hasUnsavedChanges() {
-            let context = ContextManager.sharedInstance().mainContext
-            let postService = PostService(managedObjectContext: context)
-             let status = NSLocalizedString("Saving...", comment: "Text displayed in HUD while a post is being saved as a draft.")
-            SVProgressHUD.setDefaultMaskType(.clear)
-            SVProgressHUD.show(withStatus: status)
-            postService.uploadPost(post, success: { [weak self] newPost in
-                self?.post = newPost
-                self?.createPostRevisionBeforePreview(completion: completion)
-                SVProgressHUD.dismiss()
-                }, failure: { error in
+    private func savePostBeforePreview(completion: @escaping ((String?, Error?) -> Void)) {
+        let context = ContextManager.sharedInstance().mainContext
+        let postService = PostService(managedObjectContext: context)
+        let draftStatus = NSLocalizedString("Saving...", comment: "Text displayed in HUD while a post is being saved as a draft.")
+        let publishedStatus = NSLocalizedString("Generating Preview...", comment: "Text displayed in HUD while a post is being saved.")
+        SVProgressHUD.setDefaultMaskType(.clear)
+
+        if post.hasUnsavedChanges() {
```

Comment:
Nitpicky, but maybe this could be tweaked to avoid the nested conditionals?  Something like
```
if !post.hasUnsavedChanges() {
      completion(nil, nil)
      return
}
```